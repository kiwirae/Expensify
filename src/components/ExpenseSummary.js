import React from 'react'
import { connect } from 'react-redux'
import expensesTotal from '../selectors/expensesTotal'
import selectExpenses from '../selectors/expenses'
import numeral from 'numeral'

const ExpenseSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    const formattedTotal = numeral(expensesTotal).format('$0,0.00')
    return (
        <div>
            {
                expenseCount === 0 ? <p>No expenses to view</p>
                    : <p>Viewing {expenseCount} {expenseWord} totalling {formattedTotal}</p>
            }
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: expensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary)