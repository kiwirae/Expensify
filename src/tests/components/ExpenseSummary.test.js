import React from 'react'
import ExpenseSummary from '../../components/ExpenseSummary'
import { shallow } from 'enzyme'

test('Should render Expense summary without expenses correctly', () => {
    const wrapper = shallow(<ExpenseSummary />)
    expect(wrapper).toMatchSnapshot()
})

test('Should render expense summary with only 1 expense.', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={50} />)
    expect(wrapper).toMatchSnapshot()
})

test('Should render expense summary with data.', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={3} expensesTotal={195} />)
})