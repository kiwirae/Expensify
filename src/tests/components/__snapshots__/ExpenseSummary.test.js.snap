// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expense summary without expenses correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Should render expense summary with only 1 expense. 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
